incubator/virtualcluster/pkg/syncer/resources/pod/checker.go

```diff
@@ -165,3 +242,46 @@ func (c *controller) checkPodsOfTenantCluster(clusterName string) {
 		}
 	}
 }
+
+// checkNodesOfTenantCluster checks if any orphan vNode is missed in c.clusterVNodeGCMap, which can happen if syncer
+// is restarted.
+// Note that this method can be expensive since it cannot leverage pod mccontroller informer cache. The List query
+// goes to tenant master directly. If this method causes performance issue, we should consider moving it to another
+// periodic thread with a larger check interval.
+func (c *controller) checkNodesOfTenantCluster(clusterName string) {
+	tenantClient, err := c.multiClusterPodController.GetClusterClient(clusterName)
+	if err != nil {
+		klog.Errorf("failed to create client from cluster %s config: %v", clusterName, err)
+		return
+	}
+
+	nodeList, err := tenantClient.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("failed to list vNode from cluster %s config: %v", clusterName, err)
+		return
+	}
+
+	for _, vNode := range nodeList.Items {
+		if vNode.Labels[constants.LabelVirtualNode] != "true" {
+			continue
+		}
+		func() {
```

Comment:
why do you create a anonymous function here? if you want to create a new context, just put the code inside a new `{}`.